src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -97,6 +95,43 @@
       Long encryptionKeyId,
       OpenChannelRequest.OnErrorOption onErrorOption,
       boolean isTestMode) {
+    this(
+        name,
+        dbName,
+        schemaName,
+        tableName,
+        offsetToken,
+        channelSequencer,
+        rowSequencer,
+        client,
+        encryptionKey,
+        encryptionKeyId,
+        onErrorOption,
+        client.getParameterProvider().getBlobFormatVersion(),
+        isTestMode || client.isTestMode()
+            ? new RootAllocator()
+            : client
+                .getAllocator()
+                .newChildAllocator(
+                    String.format("%s_%s", name, channelSequencer),
+                    0,
+                    client.getAllocator().getLimit()));
+  }
+
+  SnowflakeStreamingIngestChannelInternal(
+      String name,
+      String dbName,
+      String schemaName,
+      String tableName,
+      String offsetToken,
+      Long channelSequencer,
+      Long rowSequencer,
+      SnowflakeStreamingIngestClientInternal<T> client,
+      String encryptionKey,
+      Long encryptionKeyId,
+      OpenChannelRequest.OnErrorOption onErrorOption,
+      Constants.BdecVersion bdecVersion,
+      BufferAllocator allocator) {
```

Comment:
could you explain why we need a new constructor? I don't see we pass null allocator, do we?

Reply:
True we do not need three constructors. I also wanted to avoid a lot of logic in constructors and move it more into builder/factory. I removed the default constructor and made this default. In future we can consider a test builder for it.
This one is actually needed to configure the BDEC version directly in tests.
Less logic in constructors usually makes them easier to unit test and leads to less implicit dependencies that are moved to named factories.
I removed null allocator, it was used once.